src/main/java/ch/uzh/ifi/hase/soprafs23/service/JoinRequestService.java

```diff
@@ -0,0 +1,87 @@
+package ch.uzh.ifi.hase.soprafs23.service;
+
+import ch.uzh.ifi.hase.soprafs23.entity.Group;
+import ch.uzh.ifi.hase.soprafs23.entity.JoinRequest;
+import ch.uzh.ifi.hase.soprafs23.entity.User;
+import ch.uzh.ifi.hase.soprafs23.repository.JoinRequestRepository;
+import ch.uzh.ifi.hase.soprafs23.rest.dto.JoinRequestPostDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class JoinRequestService {
+
+    private final JoinRequestRepository joinRequestRepository;
+    private final GroupService groupService;
+    private final UserService userService;
+
+    @Autowired
+    public JoinRequestService(JoinRequestRepository joinRequestRepository, GroupService groupService, UserService userService) {
+        this.joinRequestRepository = joinRequestRepository;
+        this.groupService = groupService;
+        this.userService = userService;
+    }
+
+    public JoinRequest createJoinRequest(JoinRequestPostDTO joinRequestPostDTO, Long groupId) {
+        Optional<JoinRequest> existingJoinRequest = joinRequestRepository.findByGuestIdAndGroupId(joinRequestPostDTO.getGuestId(), groupId);
+        if (existingJoinRequest.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "A request could not be sent");
```

Comment:
this message could be more specific, something like "guest X already has an open request to join group Y"